common/beerocks/scripts/prplmesh_utils.sh.in

```diff
@@ -315,7 +316,22 @@ main() {
             ;;
         "start_wait")
             start_function
-            wait -n
+
+            # Store PIDs of the Controller and Agents
+            PREV_PIDS=$(pgrep "beerocks_(agent|controller)")
+            CURR_PIDS=$PREV_PIDS
+            
+            # Monitor the number of running processes
+            while [ ${#CURR_PIDS[@]} -ge ${#PREV_PIDS[@]} ]; do
+                sleep 1
+                CURR_PIDS=$(pgrep "beerocks_(agent|controller)")
+                if [ ${#CURR_PIDS[@]} -gt ${#PREV_PIDS[@]} ]; then
+                        PREV_PIDS=$CURR_PIDS
+                fi
+            done
+            
+            # If one of the main processes stopped, stop the others
+            stop_function
```

Comment:
Am I craz, or will this *always* kill the just-started processes again?

Reply:
The while loop will break only if one of the monitored processes exists.
In this case, we want to cleanup everything.
Perhaps this case can be renamed to "run_wait"

Reply:
> The while loop will break only if one of the monitored processes exists.
to clarify - @vitalybu meant "exits" instead of "exists"

Reply:
Right, now I get it, thanks!

Did you change the commit message? Because now it is clear to me and yesterday it wasn't, so *something* has changed :-)